x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline_button.test.tsx

```diff
@@ -6,69 +6,118 @@
+
+    const saveTimelineIcon = container.querySelectorAll(
+      '[data-test-subj="save-timeline-button-icon"]'
+    )[0];
+
+    fireEvent.click(saveTimelineIcon);
+
+    await waitFor(() => {
+      expect(
+        container.querySelector('[data-test-subj="save-timeline-btn-tooltip"]')
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  it('should show a button with pencil icon', () => {
+    const { container } = render(<TestSaveTimelineButton {...props} />);
     expect(
-      component.find('[data-test-subj="save-timeline-button-icon"]').first().prop('iconType')
-    ).toEqual('pencil');
+      container.querySelectorAll('[data-test-subj="save-timeline-button-icon"]')[0].firstChild
+    ).toHaveAttribute('data-euiicon-type', 'pencil');
   });
-  test('should not show a modal when showOverlay equals false', () => {
-    const component = mount(
+
+  it('should have edit timeline btn disabled with tooltip if user does not have write access', () => {
+    (useUserPrivileges as jest.Mock).mockReturnValue({
+      kibanaSecuritySolutionsPrivileges: { crud: false },
+    });
+    const { container } = render(
       <TestProviders>
         <SaveTimelineButton {...props} />
       </TestProviders>
     );
-    expect(component.find('[data-test-subj="save-timeline-modal"]').exists()).toEqual(false);
+    expect(
+      container.querySelectorAll('[data-test-subj="save-timeline-button-icon"]')[0]
+    ).toBeDisabled();
   });
-  test('should show a modal when showOverlay equals true', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(true);
-    expect(component.find('[data-test-subj="save-timeline-modal-comp"]').exists()).toEqual(false);
-    component.find('[data-test-subj="save-timeline-button-icon"]').first().simulate('click');
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(false);
-    expect(component.find('[data-test-subj="save-timeline-modal-comp"]').exists()).toEqual(true);
+
+  it('should show a modal when showOverlay equals true', async () => {
+    (useUserPrivileges as jest.Mock).mockReturnValue({
+      kibanaSecuritySolutionsPrivileges: { crud: true },
+    });
+    const { container, baseElement } = render(<TestSaveTimelineButton {...props} />);
+    expect(
+      container.querySelector('[data-test-subj="save-timeline-modal"]')
+    ).not.toBeInTheDocument();
+
+    const saveTimelineIcon = container.querySelectorAll(
+      '[data-test-subj="save-timeline-button-icon"]'
+    )[0];
+
+    fireEvent.click(saveTimelineIcon);
+
+    await waitFor(() => {
+      expect(
+        container.querySelector('[data-test-subj="save-timeline-btn-tooltip"]')
+      ).not.toBeInTheDocument();
+      expect(
+        baseElement.querySelectorAll('[data-test-subj="save-timeline-modal"]')[0]
```

Comment:
Could you explain why we need to query the baseElement here?

Reply:
@janmonschke , the tooltip is a portal element and is inserted not in the component/container but  outside. hence `baseElement` gives the root of the complete dom rendered by `rendered` function.

Reply:
Gotcha! Using `screen` we should be able to select it though, right?